packages/angular-cli/tasks/build-webpack.ts

```diff
@@ -2,41 +2,22 @@ import * as rimraf from 'rimraf';
 import * as path from 'path';
 const Task = require('../ember-cli/lib/models/task');
 import * as webpack from 'webpack';
-import { BuildOptions } from '../commands/build';
+import { BuildTaskOptions } from '../commands/build';
 import { NgCliWebpackConfig } from '../models/webpack-config';
-import { getWebpackStatsConfig } from '../models/';
+import { getWebpackStatsConfig } from '../models/webpack-configs/utils';
 import { CliConfig } from '../models/config';
 
 
 export default Task.extend({
-  run: function (runTaskOptions: BuildOptions) {
+  run: function (runTaskOptions: BuildTaskOptions) {
 
     const project = this.cliProject;
 
-    const outputDir = runTaskOptions.outputPath || CliConfig.fromProject().config.apps[0].outDir;
-    const deployUrl = runTaskOptions.deployUrl ||
```

Comment:
I transferred it over to `packages/angular-cli/models/webpack-config.ts`, in the `addAppConfigDefaults` function.

To make sure your functionality in #4090 also is in, I force a value for `deployUrl` in `packages/angular-cli/tasks/serve-webpack.ts` via `serveDefaults`.

This way in for `ng serve`, the default for `--deploy-url` is `''` instead of undefined and thus the default in `angular-cli.json` is not used.

I believe this was your intention.